WNPRC_EHR/src/org/labkey/wnprc_ehr/pages/husbandry/WaterCalendar.jsp

```diff
@@ -402,8 +402,8 @@
                                             String rowid = "";
                                             String altmeaning = "";
                                             for(JSONObject frequency : husbandryFrequencyList) {
-                                                if ( frequency.getString("altmeaning") != null && !frequency.getString("altmeaning").trim().equals("")) {
-                                                     rowid= frequency.getString("rowid");
+                                                if ( !frequency.isNull("altmeaning") && !frequency.getString("altmeaning").trim().equals("")) {
```

Comment:
Please test, but I believe this would be equivalent: `StringUtils.isNotBlank(frequency.optString("altmeaning", null))` (that's `org.apache.commons.lang3.StringUtils`)

Reply:
That looks like it would be equivalent. I do not have the setup to test this page very easily so I might leave this as is